ibm/service/power/data_source_ibm_pi_shared_processor_pool.go

```diff
@@ -0,0 +1,183 @@
+				Description: "Shared processor pool allocated cores",
+			},
+
+			Attr_SharedProcessorPoolStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the shared processor pool",
+			},
+
+			Attr_SharedProcessorPoolStatusDetail: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status details of the shared processor pool",
+			},
+
+			Attr_SharedProcessorPoolInstances: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of server instances deployed in the shared processor pool",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_SharedProcessorPoolInstanceCpus: {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Computed:    true,
+							Description: "The amount of cpus for the server instance",
+						},
+						Attr_SharedProcessorPoolInstanceUncapped: {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Computed:    true,
+							Description: "Identifies if uncapped or not",
+						},
+						Attr_SharedProcessorPoolInstanceAvailabilityZone: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "Availability zone for the server instances",
+						},
+						Attr_SharedProcessorPoolInstanceId: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "The server instance ID",
+						},
+						Attr_SharedProcessorPoolInstanceMemory: {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Computed:    true,
+							Description: "The amount of memory for the server instance",
+						},
+						Attr_SharedProcessorPoolInstanceName: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "The server instance name",
+						},
+						Attr_SharedProcessorPoolInstanceStatus: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "Status of the server",
+						},
+						Attr_SharedProcessorPoolInstanceVcpus: {
+							Type:        schema.TypeFloat,
+							Optional:    true,
+							Computed:    true,
+							Description: "The amout of vcpus for the server instance",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMPISharedProcessorPoolRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
+	poolID := d.Get(Arg_SharedProcessorPoolID).(string)
+	client := st.NewIBMPISharedProcessorPoolClient(ctx, sess, cloudInstanceID)
+
+	response, err := client.Get(poolID)
+	if err != nil || response == nil {
+		return diag.FromErr(fmt.Errorf("error fetching the shared processor pool: %s", err))
```

Comment:
```suggestion
		return diag.Errorf("error fetching the shared processor pool: %v", err)
```

Reply:
Could you please change this everywhere?

Reply:
fixed